include/ignition/gazebo/components/Model.hh

```diff
@@ -29,14 +32,64 @@ namespace gazebo
 {
 // Inline bracket to help doxygen filtering.
 inline namespace IGNITION_GAZEBO_VERSION_NAMESPACE {
+namespace serializers
+{
+  class SdfModelSerializer
+  {
+    /// \brief Serialization for `sdf::Model`.
+    /// \param[in] _out Output stream.
+    /// \param[in] _time Model to stream
+    /// \return The stream.
+    public: static std::ostream &Serialize(std::ostream &_out,
+                const sdf::Model &_model)
+    {
+      sdf::ElementPtr modelElem = _model.Element();
+      if (!modelElem)
+      {
+        ignerr << "Unable to serialize sdf::Model" << std::endl;
+        return _out;
+      }
+
+      _out << "<?xml version=\"1.0\" ?>"
+           << "<sdf version='" << SDF_PROTOCOL_VERSION << "'>"
+           << modelElem->ToString("")
+           << "</sdf>";
+      return _out;
+    }
+
+    /// \brief Deserialization for `sdf::Model`.
+    /// \param[in] _in Input stream.
+    /// \param[out] _model Model to populate
+    /// \return The stream.
+    public: static std::istream &Deserialize(std::istream &_in,
+                sdf::Model &_model)
+    {
+      sdf::Root root;
+      std::string sdf(std::istreambuf_iterator<char>(_in), {});
+
+      sdf::Errors errors = root.LoadSdfString(sdf);
+      if (!errors.empty())
```

Comment:
It may be better to check `root.Element()->GetElement("model") == nullptr` instead since errors can happen but depending on the error the model could still be loaded anyway. For example, in parameter passing if a user tries to update a parameter from an original model incorrectly then `errors` will be nonempty but those updates are skipped and the original model is still loaded (assuming no other errors occurred)

Reply:
Thanks, good point! Changed in f9e2626.